observability/alerts/tested-rules/cluster-service-slo-rules.yaml

```diff
@@ -0,0 +1,127 @@
+      labels:
+        service: "clusters-service-metrics"
+
+    - record: availability:api_inbound_request_count:burnrate3d
+      expr: |
+        sum(max without(prometheus_replica) (rate(api_inbound_request_count{namespace="clusters-service",code=~"5..", service="clusters-service-metrics"}[3d])))
+        /
+        sum(max without(prometheus_replica) (rate(api_inbound_request_count{namespace="clusters-service", service="clusters-service-metrics"}[3d])))
+      labels:
+        service: "clusters-service-metrics"
+
+  # API Latency Recording Rules
+  - name: arohcp_cs_api_latency_recording_rules
+    interval: 30s
+    rules:
+    # P99 Latency SLI (28-day)
+    - record: latency:api_inbound_request_duration:p99_sli_ratio_28d
+      expr: |
+        sum(max without(prometheus_replica) (rate(api_inbound_request_duration_bucket{namespace="clusters-service", service="clusters-service-metrics", le="1", code!~"5.."}[28d])))
+        /
+        sum(max without(prometheus_replica) (rate(api_inbound_request_count{namespace="clusters-service", service="clusters-service-metrics"}[28d])))
+      labels:
+        service: "clusters-service-metrics"
+
+    # P90 Latency SLI (28-day)
+    - record: latency:api_inbound_request_duration:p90_sli_ratio_28d
+      expr: |
+        sum(max without(prometheus_replica) (rate(api_inbound_request_duration_bucket{namespace="clusters-service", service="clusters-service-metrics", le="0.1", code!~"5.."}[28d])))
+        /
+        sum(max without(prometheus_replica) (rate(api_inbound_request_count{namespace="clusters-service", service="clusters-service-metrics"}[28d])))
+      labels:
+        service: "clusters-service-metrics"
+
+  # API Availability Alerts
+  - name: cluster-service-slos
+    rules:
+    - alert: ClustersServiceAPIAvailability5mto1hor30mto6hErrorBudgetBurn
+      annotations:
+        description: 'API is rapidly burning its 28 day availability error budget (99% SLO)'
+        runbook_url: 'https://runbooks.prometheus-operator.dev/runbooks/kubernetes/kubeapierrorbudgetburn'
+        summary: 'Cluster Service API error budget burn rate is too high'
+      expr: |
+        (
+        sum(max without(prometheus_replica) (availability:api_inbound_request_count:burnrate5m{service="clusters-service-metrics"})) > (13.44 * (1-0.99000))
+        and
+        sum(max without(prometheus_replica) (availability:api_inbound_request_count:burnrate1h{service="clusters-service-metrics"})) > (13.44 * (1-0.99000))
+        )
+        or
+        (
+        sum(max without(prometheus_replica) (availability:api_inbound_request_count:burnrate30m{service="clusters-service-metrics"})) > (5.6 * (1-0.99000))
+        and
+        sum(max without(prometheus_replica) (availability:api_inbound_request_count:burnrate6h{service="clusters-service-metrics"})) > (5.6 * (1-0.99000))
+        )
+      for: 5m
```

Comment:
would be good to get confirmation from sre team on best practice for setting this value for this type of alert.

Reply:
sounds good 👍 we can fine tune it later.